flepimop/gempyor_pkg/src/gempyor/utils.py

```diff
@@ -137,7 +158,32 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
-def search_and_import_plugins_class(plugin_file_path: str, path_prefix: str, class_name: str, **kwargs):
+def search_and_import_plugins_class(plugin_file_path: str, path_prefix: str, class_name: str, **kwargs: dict[str, any]) -> any:
+    """
+    Function serving to create a class that finds and imports the necessary modules.
+
+    Args:
+        plugin_file_path: Pathway to the module.
+        path_prefix: Pathway prefix to the module.
+        class_name: Name of the class.
+        **kwargs: Further arguments passed to initilization of the class.
+
+    Returns:
+        The instance of the class that was instantiated with provided **kwargs.
+
+    Example:
+        Suppose there is a module called `my_plugin.py with a class `MyClass` located at `/path/to/plugin/`.
+        
+        Dynamically import and instantiate the class:
+        ```
```

Comment:
Instead of using triple backticks I think using the doctest style examples would be best. See: https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html, specifically the `example_generator` function.